### ISSUE TITLE
Inconsistent Escape Behavior in ZVM Visual Mode

### ISSUE THREAD

**Issue by tummetott** [jeffreytse/zsh-vi-mode#257]
## Basic examination

- [x] I have read through the [README](https://github.com/jeffreytse/zsh-vi-mode) page
- [x] I have the latest version of zsh-vi-mode
- [x] I have tested with another terminal program 

## Problem description

Hello,
I've encountered a recurrent issue in ZVM where unintentional entry into visual mode leads to a challenging exit process. Often, by habit, I press `i` aiming to switch to insert mode, but this action instead initiates the operator pending mode within visual mode. This is an expected behavior of the mode. However, the problem arises when attempting to exit from this state. Repeatedly pressing `ESC` does not lead to an exit from the operator pending mode. It necessitates an additional action, the `w` key to select a word, before `ESC` successfully exits visual mode.

## Reproduction steps

1. Enter normal mode: Press `ESC`
2. Switch to visual mode: Press `v`
3. Activate operator pending mode: Press `i` or `a`
4. Attempt to exit operator pending mode: Press `ESC`

## Expected behavior

Upon completion of Step 4, the expectation is to exit operator pending mode directly with the `ESC` key.

**Comment by 94kai** [comment]
i have the same problem

**Comment by johnpyp** [comment]
I've had the same issue for months at this point, only just realized it was caused by this specific behavior.

**Comment by tummetott** [comment]
Well you found the right lines of code but uncommenting it is no solution. It is valid vim behavior to enter operator pending mode pressing `v` -> `a` or `i`. The problem is that ZVM does not exit this mode when pressing ESC in this state.

**Comment by johnpyp** [comment]
Yeah, I just did that as a temporary patch. I tried to debug the issue but couldn't find where the handling should happen... not familiar enough.

**Comment by sabattle** [comment]
I run into this probably at least once every time I use my terminal and it always throws me for a loop lol. Is there any update to this or does anyone have any better workarounds?

**Comment by kschaab** [comment]
If you are here like me and want to prevent this, the easiest solution which is to remove "i" and "a" binds in visual mode (note zvm_after_init is to hook after zvm loads).  
```shell
function zvm_after_init {
    zvm_bindkey visual -r "i"
    zvm_bindkey visual -r "a"
}
```

This removes the ability to use i/a in visual mode though, which wasn't important enough for me due to escape not working.  I found this to be super frustrating when there is nothing on the prompt due to accidental entry in normal mode of say `vi` without a valid operator as you need this to move forward.  With a blank prompt it is super difficult to see what is happening.  So different work around for this specific case is to hook vicmd "v" and only transition to visual mode when there is text in `$BUFFER`:

```shell
function safe-zvm-visual-mode {
    if [[ -z "$BUFFER" ]]; then
        # Buffer is empty: turn off the visual selection and force Normal mode
        REGION_ACTIVE=0
        zvm_select_vi_mode vicmd
        zle reset-prompt
    else
        # Buffer has text: execute the native Zsh visual mode widget
        zle .visual-mode
    fi
}
zle -N safe-zvm-visual-mode

function zvm_after_init {
    bindkey -M vicmd 'v' safe-zvm-visual-mode
}
```

I went with the simpler option because I do not frequently use i+operator and a+operator movements in visual mode in zsh and there are still issues with the second approach since escape doesn't work in these cases and after I hit escape it is hard to move to the mental model of completing the operator to be able to actually escape.
